gr-digital/lib/diversity_combiner_cc_impl.cc

```diff
@@ -87,20 +85,26 @@ namespace gr {
                                                uint64_t offset,
                                                uint64_t length){
       // Calculate the magnitude square of the complex CSI vector.
-      for (int i = 0; i < d_num_inputs; ++i) {
-        d_csi_squared[i] = std::norm(d_csi[i]);
+      for (int k = 0; k < d_vlen; ++k) {
+        for (int i = 0; i < d_num_inputs; ++i) {
+          d_csi_squared[k][i] = std::norm(d_csi[k][i][0]);
+        }
       }
       if(d_combining_technique.compare("SC") == 0) { // Selection combining
         // Search for path coefficient with maximum magnitude.
-        d_best_path = std::distance(d_csi_squared.begin(),
-                                    std::max_element(d_csi_squared.begin(),
-                                                     d_csi_squared.end()));
+        for (int k = 0; k < d_vlen; ++k) {
+          d_best_path[k] = std::distance(d_csi_squared[k].begin(),
+                                      std::max_element(d_csi_squared[k].begin(),
+                                                       d_csi_squared[k].end()));
+        }
       }
       else if(d_combining_technique.compare("MRC") == 0) { // Maximum-Ratio combining
         // Calculate the normalized weighting coefficients.
-        float total_path_energy = std::accumulate(d_csi_squared.begin(), d_csi_squared.end(), 0.0);
-        for (int i = 0; i < d_num_inputs; ++i) {
-          d_mrc_weighting[i] = std::polar(std::sqrt(d_csi_squared[i]/total_path_energy), -std::arg(d_csi[i]));
+        for (int k = 0; k < d_vlen; ++k){
+          float total_path_energy = std::accumulate(d_csi_squared[k].begin(), d_csi_squared[k].end(), 0.0);
+          for (int i = 0; i < d_num_inputs; ++i) {
+            d_mrc_weighting[k][i] = std::polar(std::sqrt(d_csi_squared[k][i]/total_path_energy), -std::arg(d_csi[k][i][0]));
```

Comment:
This looks a lot like it's just a fancy way of saying `d_mrc_weighting[k][i] = -d_csi[k][i][0]/sqrt(total_path_energy)`, but I'd have to write this down.

Shouldn't this be the complex conjugate of `d_csi[k][i][0]/sqrt(total_path_energy)` instead?